### PR TITLE
Added BTG 0.15.0.2 Image

### PR DESCRIPTION
**Pull request by Vutov**

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Can you remove the generated dockers? I will change soon so that the docker files are generated on the fly. 
With the increase of altcoins we would get combinasional explosion by continuing doing like I did for btc and ltc.

**Comment by Vutov** [comment]
I've removed all generated files. Is this what you meant, or to remove bgold-cli.ps1 and bgold-cli.sh as well?

File: docker-compose-generator/docker-fragments/bitcoingold.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+version: "3"
+
+services:
+  bgoldd:
```
Comment (NicolasDorier) [inline docker-compose-generator/docker-fragments/bitcoingold.yml]: why two `d`?

**Comment by NicolasDorier** [comment]
bgold-cli.ps1 and .sh are fine. You need to make them `chmod +x` though by running:

```
git update-index --chmod=+x  bgold-cli.ps1
git update-index --chmod=+x  bgold-cli.sh
```

File: docker-compose-generator/docker-fragments/bitcoingold.yml (hunk 1 above)
Comment (Vutov) [reply docker-compose-generator/docker-fragments/bitcoingold.yml]: No particular reason, the bitcoind is called bgoldd, so I kept it here for consistency.

**Comment by NicolasDorier** [comment]
So have you tried? I guess think you did everything needed to have docker on mainnet.

**Comment by Vutov** [comment]
I've tried BTCPay with local full mainnet BTG node. I've tried the BTG image and pulled and ran the kamigawabul/docker-bitcoingold/:0.15.0.2 image. I'm new to Docker, but hopefully it is working :)

**Comment by NicolasDorier** [comment]
Try the docker-compose, time well spent. [Start here](https://github.com/btcpayserver/btcpayserver-docker#for-docker-noobs-).

You can run the generator by running `generate-docker-compose` script.

**Comment by Vutov** [comment]
It is working with mainnet node using docker-compose. Just keep in mind the C# generator is putting NBXPLORER_BTCNODEENDPOINT, NBXPLORER_BTCRPCURL and BTCPAY_BTCEXPLORERURL instead of BTGNODE, BTGRPC and so on

File: docker-compose-generator/docker-fragments/bitcoingold.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          whitelist=0.0.0.0/0
+      expose:
+        - "43782"
+        - "39388"
+      volumes:
+        - "bitcoingold_datadir:/data"
+  nbxplorer:
+      environment:
+        NBXPLORER_CHAINS: "btg"
+        NBXPLORER_BTCRPCURL: http://bgoldd:43782/
```
Comment (NicolasDorier) [inline docker-compose-generator/docker-fragments/bitcoingold.yml]: then change that to `NBXPLORER_BTGRPCURL`

**Comment by NicolasDorier** [comment]
It does use this because you tell him to do this.

File: docker-compose-generator/docker-fragments/bitcoingold.yml (hunk 2 above)
Comment (Vutov) [reply docker-compose-generator/docker-fragments/bitcoingold.yml]: Oh, I'm stupid. Thanks

**Comment by Vutov** [comment]
Fixed

**Comment by NicolasDorier** [comment]
Very good! I keep that open for now, will merge it. I need to work on the `entrypoint.sh` of azure so I provide BTG in Azure.

**Comment by Vutov** [comment]
Great, thanks!

**Comment by NicolasDorier** [comment]
Ok I made it to make the docker-compose-generator more flexible, can you check https://github.com/btcpayserver/btcpayserver-docker#how-to-extend-with-my-own-crypto and update this PR?

I will then test it.

I am doing that so that the one click deploy on Azure will propose altcoins to support in a dropdown list.

**Comment by Vutov** [comment]
Sure, I'll test it tomorrow and update the PR. Keep you posted.

File: docker-compose-generator/src/Program.cs [hunk 3]
```diff
@@ -27,20 +27,24 @@ static void Main(string[] args)
...
+							        string name = $"{btc}-{ltc}-{btg}-{lightning}".Replace("--", "-");
+							        if (name.EndsWith("-"))
+							            name = name.Substring(0, name.Length - 1);
+							        if (name.StartsWith("-"))
+							            name = name.Substring(1, name.Length - 1);
+							        var composition = new DockerComposition();
+							        composition.SelectedCryptos = new HashSet<string>();
+							        composition.SelectedCryptos.Add(btc);
+							        composition.SelectedCryptos.Add(ltc);
+							        composition.SelectedCryptos.Add(btg);
```
Comment (NicolasDorier) [inline docker-compose-generator/src/Program.cs]: remove `btg` the pregen is only for btc and ltc for legacy reason.

**Comment by NicolasDorier** [comment]
very cool. Now I need to update the azure deployment. Will do.
Just a small fix: Do not modify the pregen.

**Comment by Vutov** [comment]
Updated the PR to keep Program.cs as is.

**Comment by NicolasDorier** [comment]
Renamed bitcoin gold to bgold.

**Comment by NicolasDorier** [comment]
BTG can now be deployed on Azure https://github.com/btcpayserver/btcpayserver-azure

**Pull request merged**